test/remote/README.md

```diff
@@ -0,0 +1,240 @@
+# End-to-end tests against a real cluster
+
+Using the tools in this directory, you can run the End-to-End testsuite against
+a real Argo CD workload, that is deployed to a K8s cluster, instead of running
+it against a locally running Argo CD.
+
+Since e2e tests are destructive, do **not** run it against an installation that
+you depend on.
+
+## Preparations
+
+### Install the Argo CD you want to test
+
+It is recommended to install in the `argocd-e2e` namespace:
+
+```shell
+kubectl create ns argocd-e2e
+kubectl -n argocd-e2e apply -f <your Argo CD installation manifests>
+```
+
+### Give the Argo CD the appropriate RBAC permissions
+
+```shell
+# If you installed to a different namespace, set accordingly
+export NAMESPACE=argocd-e2e
+# If you installed Argo CD via Operator, set accordingly
+# export ARGOCD_E2E_NAME_PREFIX=argocd-cluster
+./test/remote/generate-permissions.sh | kubectl apply -f -
+```
+
+### Build the repository container image
+
+You will need to build & publish a container image that will hold the required
+testing repositories.
```

Comment:
Do you mean, code repositories?

Reply:
No, the repositories hold the manifests used during testing. But if that is misleading, we could rephrase it to "the required repositories that hold our test manifests" - would that be better understandable?